### PR TITLE
🪟 🎉 Add auto-detect schema changes option to connection UIs

### LINKED ISSUE #17826: Add auto-detect schema changes option to create connection and replication settings pages
Add the option  to "pause sync" or "ignore" to the connection create form and replication settings. Ensure that the options update correctly.

The prototype contains an option for "Reset" / "Accept changes & Ignore Data". This is out of scope and should be ignored.

See "PAUSE_SYNC" and "IGNORE" flows in prototype:

https://www.figma.com/proto/liLQhcbpVHiEDW18kiXQe3/01_03_CONNECTIONS?node-id=973%3A41498&scaling=min-zoom&page-id=972%3A36208&starting-point-node-id=973%3A41498&show-proto-sidebar=1

### PR DESCRIPTION
**Pull request by teallarson**
## What
closes #17826

Screenshot 2022-11-29 at 3 37 35 PM

## How
Adds a new field using a similar pattern to the `schedule` field.

I've made a note to add E2E coverage in #17936

## Recommended reading order
1. `NonBreakingChangesPreferenceField.*`
2. `ConnectionFormFields.tsx`
3.  other

## Note:
As of right now, the API on `frontend-dev` has not been updated to include all of these changes.  I will update this comment when `frontend-dev` is sorted out.  However, this PR can be tested against OSS as it stands with `REACT_APP_AUTO_DETECT_SCHEMA_CHANGES=true`

### CODE REVIEW

File: airbyte-webapp/src/locales/en.json [hunk 1]
```diff
@@ -148,6 +148,10 @@
   "connectionForm.defaultSyncMode.message": "Set how Airbyte reads from the source and writes to the destination, across all streams <b>when the option is available</b>.",
   "connectionForm.bulkEdit.cancel": "Cancel",
   "connectionForm.bulkEdit.apply": "Apply",
+  "connectionForm.nonBreakingChangesPreference.label": "Non-breaking schema updates detected*",
+  "connectionForm.nonBreakingChangesPreference.message": "When Airbyte detects that the schema of the source has changed, what would you like to do?",
```
Comment (tealjulia) [inline airbyte-webapp/src/locales/en.json:152]: @andyjih @edmundito I did not see any helper text written out in the Figma.  I winged it, but it's verbose.  Suggestions on trimming this down?

Comment (edmundito) [reply airbyte-webapp/src/locales/en.json:152]: Here's what I got based on how the description for the setting above is worded:

"Set how Airbyte handles syncs when it detects a non-breaking schema change in the source"

File: airbyte-webapp/src/views/Connection/ConnectionForm/formConfig.tsx [hunk 2]
```diff
@@ -126,6 +128,10 @@ export const createConnectionValidationSchema = ({
             .defined("form.empty.error"),
         });
       }),
+      nonBreakingChangesPreference: yup
+        .string()
+        .oneOf([NonBreakingChangesPreference.disable, NonBreakingChangesPreference.ignore])
+        .required("form.empty.error"),
```
Comment (edmundito) [inline airbyte-webapp/src/views/Connection/ConnectionForm/formConfig.tsx:134]: This is a big tricky but there needs to be a check if the feature is enabled.

File: airbyte-webapp/src/views/Connection/ConnectionForm/components/NonBreakingChangesPreferenceField.module.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
+@use "scss/colors";
+@use "scss/variables";
+
+.flexRow {
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: flex-start;
+  gap: 10px;
```
Comment (edmundito) [inline airbyte-webapp/src/views/Connection/ConnectionForm/components/NonBreakingChangesPreferenceField.module.scss:9]: This and the margin below can use the scss spacing vars

File: airbyte-webapp/src/views/Connection/ConnectionForm/components/NonBreakingChangesPreferenceField.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+          nextLine
+          label={formatMessage({
+            id: "connectionForm.nonBreakingChangesPreference.label",
+          })}
+          message={formatMessage({
+            id: "connectionForm.nonBreakingChangesPreference.message",
+          })}
+        />
+      </div>
+      <div className={styles.rightFieldCol} style={{ pointerEvents: mode === "readonly" ? "none" : "auto" }}>
```
Comment (edmundito) [inline airbyte-webapp/src/views/Connection/ConnectionForm/components/NonBreakingChangesPreferenceField.tsx:42]: The pointerEvents style could be set through a className?

**Review by edmundito — CHANGES_REQUESTED** [review]

**Review by edmundito — APPROVED** [review]
Tested locally for create connection, edit replication settings, refresh source schema, and cancel edit.

**Review by krishnaglick — APPROVED** [review]

**Pull request merged**